src/NServiceBus.Core/Routing/MessageDrivenSubscriptions/MessageDrivenUnsubscribeTerminator.cs

```diff
@@ -27,24 +31,63 @@ public override void Terminate(UnsubscribeContext context)
 
             if (!publisherAddresses.Any())
             {
-                throw new Exception(string.Format("No destination could be found for message type {0}. Check the <MessageEndpointMappings> section of the configuration of this endpoint for an entry either for this specific message type or for its assembly.", eventType));
+                throw new Exception($"No destination could be found for message type {eventType}. Check the <MessageEndpointMappings> section of the configuration of this endpoint for an entry either for this specific message type or for its assembly.");
             }
 
+            var unsubscribeTasks = new List<Task>();
             foreach (var publisherAddress in publisherAddresses)
             {
                 Logger.Debug("Unsubscribing to " + eventType.AssemblyQualifiedName + " at publisher queue " + publisherAddress);
 
-                var subscriptionMessage = ControlMessageFactory.Create(MessageIntentEnum.Unsubscribe);
+                var unsubscribeMessage = ControlMessageFactory.Create(MessageIntentEnum.Unsubscribe);
 
-                subscriptionMessage.Headers[Headers.SubscriptionMessageType] = eventType.AssemblyQualifiedName;
-                subscriptionMessage.Headers[Headers.ReplyToAddress] = replyToAddress;
+                unsubscribeMessage.Headers[Headers.SubscriptionMessageType] = eventType.AssemblyQualifiedName;
+                unsubscribeMessage.Headers[Headers.ReplyToAddress] = replyToAddress;
 
+                var address = publisherAddress;
 
-                var dispatchOptions = new DispatchOptions(new DirectToTargetDestination(publisherAddress), context);
-                dispatcher.Dispatch(new [] { new TransportOperation(subscriptionMessage, dispatchOptions)}).GetAwaiter().GetResult();
+                unsubscribeTasks.Add(SendUnsubscribeMessageWithRetries(address, unsubscribeMessage, eventType.AssemblyQualifiedName, context));
             }
+
+            return Task.WhenAll(unsubscribeTasks.ToArray());
         }
 
+        async Task SendUnsubscribeMessageWithRetries(string destination, OutgoingMessage unsubscribeMessage, string messageType, ContextBag context, int retriesCount = 0)
```

Comment:
Ok, I think that makes sense for this pull. That said us doing retries for some operations feels vierd. Would be consider it for bus.Send?

I think we need to discuss how to handle this long term separately